Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonConfigHelper.cs

```diff
@@ -415,6 +414,168 @@ public void ForceRefresh()
+
+            if (iconSet != defaultIconSet)
+            {  // This button is using a custom icon set, so we can't assume material differences mean it needs an upgrade.
+                return;
+            }
+
+            if (iconQuadRenderer.sharedMaterial == defaultButtonQuadMaterial) 
+            {   // This button is using the default material, so it's not a customized button.
+                return;
+            }
+
+            Material targetQuadMaterial = iconQuadRenderer.sharedMaterial;
+            Texture targetQuadIcon = targetQuadMaterial.mainTexture;
+
+            if (targetQuadIcon == null)
+            {   // There is no icon to copy.
+                return;
+            }
+
+            Debug.Log("Existing quad icon is " + targetQuadIcon.name);
+
+            ButtonIconSet targetIconSet = iconSet;
+            bool createdIconSet = false;
+            string generatedIconSetFolder = System.IO.Path.Combine(MixedRealityToolkitFiles.GetGeneratedFolder, customIconSetsFolderName);
+
+            // If this icon set doesn't have our icon in it, we need to either add it or create a new icon set
+            if (!iconSet.TryGetQuadIcon(targetQuadIcon.name, out Texture2D quadIcon))
+            {
+                if (iconSet == defaultIconSet)
+                {   // If we're using the default icon set, we have to create a new set to add the icon
+                    if (!AssetDatabase.IsValidFolder(generatedIconSetFolder))
+                    {   // Create the folder if it doesn't exist
+                        AssetDatabase.CreateFolder(MixedRealityToolkitFiles.GetGeneratedFolder, customIconSetsFolderName);
+                    }
+
+                    string generatedIconSetPath = System.IO.Path.Combine(generatedIconSetFolder, generatedIconSetName + ".asset");
+                    targetIconSet = (ButtonIconSet)AssetDatabase.LoadAssetAtPath(generatedIconSetPath, typeof(ButtonIconSet));
+
+                    if (targetIconSet == null)
+                    {   // If the icon set doesn't already exist, duplicate the default
+                        ScriptableObject duplicateIconSet = Instantiate<ButtonIconSet>(defaultIconSet);
+                        duplicateIconSet.name = generatedIconSetName;
+
+                        AssetDatabase.CreateAsset(duplicateIconSet, generatedIconSetPath);
+                        AssetDatabase.SaveAssets();
+                        AssetDatabase.Refresh(ImportAssetOptions.ForceUpdate);
+
+                        targetIconSet = (ButtonIconSet)AssetDatabase.LoadAssetAtPath(generatedIconSetPath, typeof(ButtonIconSet));
+                        createdIconSet = true;
+                    }
+                }
+            }
+
+            bool selectIconSet = false;
+            if (createdIconSet)
+            {
+                selectIconSet = EditorUtility.DisplayDialog("Custom Icon Set Created", string.Format(customIconSetCreatedMessage, generatedIconSetFolder), "View Asset", "OK");
```

Comment:
It might be good to point people to the button configurator documentation here, or explain that the custom icon set is used by the Button Configurator component. Otherwise people may be confused what that message means.

Reply:
Done.